Std/Data/BitVec/Lemmas.lean

```diff
@@ -293,6 +293,9 @@ theorem not_def {x : BitVec v} : ~~~x = allOnes v ^^^ x := rfl
     BitVec.toNat (x <<< n) = BitVec.toNat x <<< n % 2^v :=
   BitVec.toNat_ofNat _ _
 
+@[simp] theorem toFin_shiftLeft {n : Nat} (x : BitVec v) :
+    BitVec.toFin (x <<< n) = Fin.ofNat' (x.toNat <<< n) (Nat.two_pow_pos v) := rfl
+
```

Comment:
It seems that shifting is defined on `Fin` as well, [in core](https://github.com/leanprover/lean4/blob/5cc0c3f145ffa623a617ca31387d7f589802c1d3/src/Init/Data/Fin/Basic.lean#L71-L72), but only as Fin -> Fin -> Fin.

An alternative rhs could be `x.toFin <<< (Fin.ofNat' n (Nat.two_pow_pos _))`, but this is not a def-eq anymore. We might want to mention this in the PR description, and use the def-eq as argument for why we choose to shift on Nats in the rhs.